### PR TITLE
Initial functionality

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR delivers the basic functionality for this role:
*  Fetch the Assessor Portal tarball from our S3 bucket
*  Extract it to an appropriate location (`/opt/assessor-portal`)

## 💭 Motivation and context ##

Eventually, this role will deploy the Assessor Portal code from a GitHub repository, but for now, we are fine with pulling a tarball out of a bucket.

This is part of the work needed for https://github.com/cisagov/cool-system/issues/202.

## 🧪 Testing ##

I ran `AWS_PROFILE=test-ansible-role-assessor-portal molecule test` with great success and all pre-commit checks passed.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [x] Tests have been added and/or modified to cover the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,7 +58,10 @@ None.
 
 ## Role Variables ##
 
-None.
+* `bucket_name` - The name of the AWS S3 bucket where the Assessor Portal
+  tarball is stored.  Defaults to `cisa-cool-third-party-production`.
+* `tarball_object_name` - The name of the AWS S3 object that is the Assessor
+  Portal tarball.  Defaults to `assessor-portal-python-v1.1.1_test.tgz`.
 
 ## Dependencies ##
```
Comment (mcdonnnj) [inline README.md:66]: You have [cisagov/ansible-role-docker](https://github.com/cisagov/ansible-role-docker) as a dependency in `meta/main.yml` but I do not see it in the README.

File: terraform/variables.tf [hunk 2]
```diff
@@ -10,6 +10,34 @@ variable "aws_region" {
...
+  default     = "cisa-cool-third-party-production"
+}
+
+variable "production_objects" {
+  type        = list(string)
+  description = "The Assessor Portal tarball object inside the production bucket."
+  default = [
+    "assessor-portal-python-v1.1.1_test.tgz",
+  ]
+}
```
Comment (jsf9k) [inline terraform/variables.tf]: You should either change the variable name and type to make it singular or modify the description to indicate plurality.

File: terraform/variables.tf [hunk 3]
```diff
@@ -10,6 +10,34 @@ variable "aws_region" {
...
+  default     = "cisa-cool-third-party-staging"
+}
+
+variable "staging_objects" {
+  type        = list(string)
+  description = "The Assessor Portal tarball object inside the staging bucket."
+  default = [
+    "assessor-portal-python-v1.1.1_test.tgz",
+  ]
+}
```
Comment (jsf9k) [inline terraform/variables.tf]: You should either change the variable name and type to make it singular or modify the description to indicate plurality.

File: terraform/outputs.tf [hunk 4]
```diff
@@ -1,20 +1,30 @@
...
-  description = "The IAM role that the CI user can assume to read SSM parameters in the staging account."
+  value       = module.staging_bucket_access.role
+  description = "The IAM role that can read the specified objects from the specified S3 staging bucket."
 }
 
 output "user" {
-  value       = module.user.user
-  description = "The CI IAM user created by this module."
+  value       = aws_iam_user.user
+  description = "The IAM user being created to test the cisagov/ansible-assessor-portal Ansible role."
```
Comment (jsf9k) [inline terraform/outputs.tf]:
```suggestion
  description = "The IAM user being created to test the cisagov/ansible-role-assessor-portal Ansible role."
```

File: terraform/outputs.tf [hunk 5]
```diff
@@ -1,20 +1,30 @@
...
+}
+
+output "user_access_key" {
+  value       = aws_iam_access_key.key
+  description = "The access key for the IAM user being created to test the cisagov/ansible-assessor-portal Ansible role."
```
Comment (jsf9k) [inline terraform/outputs.tf]:
```suggestion
  description = "The access key for the IAM user being created to test the cisagov/ansible-role-assessor-portal Ansible role."
```

**Review by jsf9k — APPROVED** [review]
I pointed out a few minor thangs that may need adjusting, but overall this looks great.  _Herculean_ work, @dav3r!

File: terraform/providers.tf [hunk 6]
```diff
@@ -9,12 +9,12 @@ provider "aws" {
   region = var.aws_region
 }
 
-# The provider used to create the role that can be assumed to do
-# everything the CI user needs to do in the staging account.
+# The provider used to create roles that can read certificates from a
+# production S3 bucket
```
Comment (mcdonnnj) [inline terraform/providers.tf]: I think this comment should be updated to reflect what we _are_ reading.

File: terraform/providers.tf [hunk 7]
```diff
@@ -23,40 +23,12 @@ provider "aws" {
   region = var.aws_region
 }
 
-# The provider used to create the role that can be assumed to do
-# everything the CI user needs to do in the production account.
+# The provider used to create roles that can read certificates from a
+# staging S3 bucket
```
Comment (mcdonnnj) [inline terraform/providers.tf]: Same suggestion here.

**Review by mcdonnnj — APPROVED** [review]
Strong work! 💪💪💪 I just had some suggestions about documentation for your consideration.

File: terraform/variables.tf (hunk 2 above)
Comment (dav3r) [reply terraform/variables.tf]: I was being lazy!  Resolved in https://github.com/cisagov/ansible-role-assessor-portal/pull/3/commits/d86a98446ebf9b90b6a314befeb6384a7d46a399.

File: terraform/variables.tf (hunk 3 above)
Comment (dav3r) [reply terraform/variables.tf]: Resolved in https://github.com/cisagov/ansible-role-assessor-portal/pull/3/commits/d86a98446ebf9b90b6a314befeb6384a7d46a399.

File: terraform/providers.tf (hunk 6 above)
Comment (dav3r) [reply terraform/providers.tf]: Good catch - fixed in https://github.com/cisagov/ansible-role-assessor-portal/pull/3/commits/4eb45c886c7b594b216996ff2e00db053b9f03ae.

File: terraform/providers.tf (hunk 7 above)
Comment (dav3r) [reply terraform/providers.tf]: Fixed in https://github.com/cisagov/ansible-role-assessor-portal/pull/3/commits/4eb45c886c7b594b216996ff2e00db053b9f03ae.

File: README.md (hunk 1 above)
Comment (dav3r) [reply README.md:66]: 🦅  👀 ❗ 
Added in https://github.com/cisagov/ansible-role-assessor-portal/pull/3/commits/6db6b7fa2c970d3e72c62422d7b6c3e87528540e.

**Pull request merged**